### PR TITLE
New: Add Box.DOMEventDelegate (fixes #47)

### LINKED ISSUE #47: Create a way for services to autowire up events like modules
People seem to really like using the autowiring of events in modules and behaviors, and when they get to modules, they are disappointed that they need to wire up their own event handlers the old-fashioned way. It would be nice to provide some facility for services to do declarative event handling in the same way behaviors and modules do.

### PR DESCRIPTION
**Pull request by nzakas**
Not yet ready for merging, just a first pass. Will update when it's ready to merge.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @nzakas has signed the CLA. Thanks for the pull request!

**Comment by j3tan** [comment]
looks good, any examples of how to mix this into services/modules?

**Comment by nzakas** [comment]
Not yet.

**Comment by nzakas** [comment]
This code is now ready for review. I've refactored `application.js` to use `DOMEventDelegate`, and in so doing, was able to close #63 (cc @mattbasta, who originally reported this internally way back when)

File: lib/dom-event-delegate.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+ * DOM element.
+ * @author Box
+ */
+
+Box.DOMEventDelegate = (function() {
+
+	'use strict';
+
+	// Supported events for modules. Only events that bubble properly can be used in T3.
+	var eventTypes = ['click', 'mouseover', 'mouseout', 'mousedown', 'mouseup',
```
Comment (j3tan) [inline lib/dom-event-delegate.js]: should we use EVENT_TYPES for this? Or do we see constants as non-arrays.

File: tests/application-test.js [hunk 2]
```diff
@@ -551,6 +551,35 @@ describe('Box.Application', function() {
 
 			});
 
+
+			it('should not be called on behaviors in when stopImmediatePropagation() is called', function() {
```
Comment (j3tan) [inline tests/application-test.js]: I'd like to add another test for when behavior1 calls stopImmediatePropagation() and causes behavior2 to not fire

**Comment by j3tan** [comment]
Looks great!

File: lib/dom-event-delegate.js (hunk 1 above)
Comment (nzakas) [inline lib/dom-event-delegate.js]: I can change it. This was the way it was in application.js, so I mindlessly copied. Probably should be all uppercase.

**Pull request merged**